moveit_simple/test/moveit_simple.cpp

```diff
@@ -50,19 +54,33 @@ TEST(MoveitSimpleTest, add_trajectory)
 {
   moveit_simple::Robot robot(ros::NodeHandle(), "robot_description", "manipulator");
   const std::string TRAJECTORY_NAME("traj1");
+  const Eigen::Affine3d pose = Eigen::Affine3d::Identity(); 
 
   ROS_INFO_STREAM("Testing loading of unknown point, should fail");
-  EXPECT_FALSE(robot.addTrajPoint("bad_traj", "unknown_name", 1.0));
+  EXPECT_THROW(robot.addTrajPoint("bad_traj", "unknown_name", 1.0),std::invalid_argument);
+  EXPECT_THROW(robot.addTrajPoint(TRAJECTORY_NAME, pose, "random_link", 5.0, "unknown_name"), tf2::TransformException);
 
   ros::Duration(2.0).sleep();  //wait for tf tree to populate
   ROS_INFO_STREAM("Testing trajectory adding of points");
-  EXPECT_TRUE(robot.addTrajPoint(TRAJECTORY_NAME, "home",      0.5));
-  EXPECT_TRUE(robot.addTrajPoint(TRAJECTORY_NAME, "waypoint1", 1.0));
-  EXPECT_TRUE(robot.addTrajPoint(TRAJECTORY_NAME, "tf_pub1",   2.0));
-  EXPECT_TRUE(robot.addTrajPoint(TRAJECTORY_NAME, "waypoint2", 3.0));
-  EXPECT_TRUE(robot.addTrajPoint(TRAJECTORY_NAME, "waypoint3", 4.0));
+  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "home",      0.5));
+  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "home",      0.5));
+  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "waypoint1", 1.0));
+  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "tf_pub1",   2.0));
+  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "waypoint2", 3.0));
+  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, "waypoint3", 4.0));
+  EXPECT_NO_THROW(robot.addTrajPoint(TRAJECTORY_NAME, pose, "tool0", 5.0, "random_name"));
+  EXPECT_NO_THROW(robot.execute(TRAJECTORY_NAME));
+
+  EXPECT_NO_THROW(robot.addTrajPoint("traj2", "waypoint4", 4.5));
+  EXPECT_THROW(robot.execute("traj2"), moveit_simple::IKFailException);
+
+// Not sure if it is a good test for Execution Failure
```

Comment:
Let's take this test out.  It really just takes advantage of a bug.  If you can't think of another way to test, that OK.